nginx-controller/nginx/configurator.go

```diff
@@ -68,6 +68,56 @@ func (cnf *Configurator) addOrUpdateIngress(ingEx *IngressEx) {
 	cnf.ingresses[name] = ingEx
 }
 
+// AddOrUpdateIngress adds or updates NGINX configuration for the Ingress resources with Mergeable Types
+func (cnf *Configurator) AddOrUpdateMergableIngress(ingExs []IngressEx) error {
+	cnf.addOrUpdateMergableIngress(ingExs, ingExs[0].Ingress.Spec.Rules[0].Host)
```

Comment:
let's have the name of master Ingress config as the name of the config for a Regular Ingress for now. This naming convention is used in Configurator and Controller in many places:
* When an Ingress resource is deleted -- https://github.com/diazjf/kubernetes-ingress/blob/349bcded25e0011a96cafff7d275d7161a3da3ac/nginx-controller/controller/controller.go#L675
* To keep track of which Ingress resources are handled by the Ingress controller -- https://github.com/diazjf/kubernetes-ingress/blob/349bcded25e0011a96cafff7d275d7161a3da3ac/nginx-controller/nginx/configurator.go#L33 This in turn is used in https://github.com/diazjf/kubernetes-ingress/blob/349bcded25e0011a96cafff7d275d7161a3da3ac/nginx-controller/controller/controller.go#L352 and https://github.com/diazjf/kubernetes-ingress/blob/349bcded25e0011a96cafff7d275d7161a3da3ac/nginx-controller/controller/controller.go#L618